File: 001-two-sum.py
class Solution:
    def __init__(self):
        self.num_to_index = {}

    def twoSum(self, nums, target):
        for i, num in enumerate(nums):
            complement = target - num
            if complement in self.num_to_index:
                return [self.num_to_index[complement], i]
            self.num_to_index[num] = i

        return None

# Example usage:
nums = [2, 7, 11, 15]
target = 9   

result = Solution().twoSum(nums, target)
print(result)  # Output: [0, 1]

File: 001-two-sum.js
function twoSum(nums, target) {
    let numObj = {};
    for (let i = 0; i < nums.length; i++) {
        let complement = target - nums[i];
        console.log(numObj[complement], complement);
        if (numObj[complement] !== undefined) {
            return [numObj[complement], i];
        }
        numObj[nums[i]] = i;
    }
}
nums = [2, 7, 11, 15]
target = 9
console.log(twoSum(nums, target));
